### ISSUE TITLE
Linter does not run on files that are excluded from build by go tags

### ISSUE THREAD

**Issue by gucio321** [golangci/golangci-lint#5080]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest [releases](https://github.com/golangci/golangci-lint/releases). Only such installations are supported.
- [X] Yes, I've searched similar [issues on GitHub](https://github.com/golangci/golangci-lint/issues) and didn't find any.
- [X] Yes, I've read the `typecheck` section of the [FAQ](https://golangci-lint.run/welcome/faq/#why-do-you-have-typecheck-errors).
- [X] Yes, I've tried with the standalone [linter](https://golangci-lint.run/usage/linters/) if available (e.g., gocritic, go vet, etc.).
- [X] I agree to follow this project's [Code of Conduct](https://github.com/golangci/golangci-lint?tab=coc-ov-file)

### Description of the problem

Hi
in [cimgui-go](https://github.com/AllenDang/cimgui-go) we have `dummy.go` files like that

```go
// +build required

// Package dummy prevents go tooling from stripping the c dependencies.
package dummy

```

(they are required to baypass go vendor system, doesn't matter here :smile:)
the file provided above is malformated (`gofmt -s -w dummy.go` will change it). 
I have `gofmt` linter enabled in my golangci.yaml but it does not detect that error.

I suppose that the reason fro that is that most linters will not run on files that are not buildable (this file for example will result in +/- CGO crash) but `gofmt` is just a text analyzer so it could be executed on such a files

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
golangci-lint has version 1.61.0 built with go1.23.2 from a1d6c56 on 2024-10-19T12:42:13Z
```

</details>

### Configuration

<details>

```console
# paste configuration file or CLI flags here
```

</details>

### Go environment

<details>

```console
$ go version && go env
go version go1.23.2 linux/amd64 GO111MODULE='' GOARCH='amd64' GOBIN='' GOCACHE='/home/gucio321/.cache/go-build' GOENV='/home/gucio321/.config/go/env' GOEXE='' GOEXPERIMENT='' GOFLAGS='' GOHOSTARCH='amd64' GOHOSTOS='linux' GOINSECURE='' GOMODCACHE='/home/gucio321/go/pkg/mod' GONOPROXY='' GONOSUMDB='' GOOS='linux' GOPATH='/home/gucio321/go' GOPRIVATE='' GOPROXY='https://proxy.golang.org,direct' GOROOT='/usr/lib/go' GOSUMDB='sum.golang.org' GOTMPDIR='' GOTOOLCHAIN='auto' GOTOOLDIR='/usr/lib/go/pkg/tool/linux_amd64' GOVCS='' GOVERSION='go1.23.2' GODEBUG='' GOTELEMETRY='local' GOTELEMETRYDIR='/home/gucio321/.config/go/telemetry' GCCGO='gccgo' GOAMD64='v1' AR='ar' CC='gcc' CXX='g++' CGO_ENABLED='1' GOMOD='/dev/null' GOWORK='' CGO_CFLAGS='-O2 -g' CGO_CPPFLAGS='' CGO_CXXFLAGS='-O2 -g' CGO_FFLAGS='-O2 -g' CGO_LDFLAGS='-O2 -g' PKG_CONFIG='pkg-config' GOGCCFLAGS='-fPIC -m64 -pthread -Wl,--no-gc-sections -fmessage-length=0 -ffile-prefix-map=/tmp/go-build917246150=/tmp/go-build -gno-record-gcc-switches'
```

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
level=info msg="golangci-lint has version 1.61.0 built with go1.23.2 from a1d6c56 on 2024-10-19T12:42:13Z"
level=info msg="[config_reader] Config search paths: [./ /home/gucio321/git/cimgui-go /home/gucio321/git /home/gucio321 /home /]"
level=info msg="[config_reader] Used config file .golangci.yml"
level=info msg="[lintersdb] Active 1 linters: [gofmt]"
level=info msg="[loader] Go packages loading at mode 7 (name|files|compiled_files) took 504.959877ms"
level=info msg="[runner/filename_unadjuster] Pre-built 0 adjustments in 158.230947ms"
level=info msg="[linters_context/goanalysis] analyzers took 0s with no stages"
level=info msg="[runner] Issues before processing: 7021, after processing: 0"
level=info msg="[runner] Processors filtering stat (in/out): cgo: 7021/0"
level=info msg="[runner] processing took 214.329µs with stages: cgo: 188.502µs, nolint: 20.994µs, max_same_issues: 904ns, filename_unadjuster: 432ns, invalid_issue: 431ns, identifier_marker: 427ns, sort_results: 343ns, skip_files: 327ns, skip_dirs: 272ns, fixer: 223ns, max_from_linter: 205ns, path_prettifier: 201ns, uniq_by_line: 191ns, exclude: 158ns, exclude-rules: 150ns, source_code: 116ns, diff: 87ns, autogenerated_exclude: 81ns, path_shortener: 74ns, path_prefixer: 73ns, max_per_file_from_linter: 73ns, severity-rules: 65ns"
level=info msg="[runner] linters took 22.016113ms with stages: goanalysis_metalinter: 21.731129ms"
level=info msg="File cache stats: 0 entries of total size 0B"
level=info msg="Memory: 8 samples, avg is 38.9MB, max is 76.6MB"
level=info msg="Execution took 687.6964ms"

```

</details>

### A minimal reproducible example or link to a public repository

<details>

```go
// +build required

// Package dummy prevents go tooling from stripping the c dependencies.
package dummy

```

</details>

### Validation

- [X] Yes, I've included all information above (version, config, etc.).

### Supporter

- [ ] I am a sponsor/backer through [GitHub](https://github.com/sponsors/golangci/) or [OpenCollective](https://opencollective.com/golangci-lint)

**Comment by ldez** [comment]
Hello,

golangci-lint follows build tags, you need to define them inside the configuration.

```yml
run:
  # List of build tags, all linters use it.
  # Default: []
  build-tags:
    - mytag
```

https://golangci-lint.run/usage/configuration/#run-configuration
